src/libsyntax/tokenstream.rs

```diff
@@ -331,35 +328,89 @@ impl TokenTree {
 /// struct itself shouldn't be directly manipulated; the internal structure is not stable,
 /// and may be changed at any time in the future. The operators will not, however (except
 /// for signatures, later on).
-#[derive(Eq,Clone,Hash,RustcEncodable,RustcDecodable)]
-pub struct TokenStream {
-    pub span: Span,
-    pub tts: Vec<TokenTree>,
+#[derive(Eq,Hash,RustcEncodable,RustcDecodable)]
+pub enum TokenStream {
```

Comment:
It's probably worth making TokenStream a struct with a single field which is this enum (and is private), since we don't want users to match/see the variants.